pkg/services/object/server.go

```diff
@@ -1928,18 +1935,34 @@ func (s *server) processSearchRequest(ctx context.Context, req *protoobject.Sear
 	default:
 	}
 
+	ttl := req.MetaHeader.GetTtl()
+	if ttl == 0 {
```

Comment:
This (and filter management) should be handled before any network interactions (like container getter above).